pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -86,19 +90,29 @@ public int submit(long requestId, QueryPlan queryPlan, long timeoutMs)
           String host = serverInstance.getHostname();
           int servicePort = serverInstance.getQueryServicePort();
           DispatchClient client = getOrCreateDispatchClient(host, servicePort);
-          Worker.QueryResponse response = client.submit(Worker.QueryRequest.newBuilder().setStagePlan(
+          Future<Worker.QueryResponse> response = client.submit(Worker.QueryRequest.newBuilder().setStagePlan(
```

Comment:
Which pool of threads is used here for the async submission ?

Reply:
no thread is used. it is sequential. e.g. clientFutureStub.submit returns immediately with a Future that will be filled by GRPC executor thread in the future